src/System.Runtime.Extensions/ref/System.Runtime.Extensions.cs

```diff
@@ -8,14 +8,93 @@
+        public System.Reflection.Assembly[] GetAssemblies() { throw null; }
+        [Obsolete("AppDomain.GetCurrentThreadId has been deprecated because it does not provide a stable Id when managed threads are running on fibers (aka lightweight threads). To get a stable identifier for a managed thread, use the ManagedThreadId property on Thread.  http://go.microsoft.com/fwlink/?linkid=14202", false)]
+        public static int GetCurrentThreadId() { throw null; }
+        public object GetData(string name) { throw null; }
+        public System.Nullable<bool> IsCompatibilitySwitchSet(string value) { throw null; }
+        public bool IsDefaultAppDomain() { throw null; }
+        public bool IsFinalizingForUnload() { throw null; }
+        public System.Reflection.Assembly Load(byte[] rawAssembly) { throw null; }
+        public System.Reflection.Assembly Load(byte[] rawAssembly, byte[] rawSymbolStore) { throw null; }
+        public System.Reflection.Assembly Load(System.Reflection.AssemblyName assemblyRef) { throw null; }
+        public System.Reflection.Assembly Load(string assemblyString) { throw null; }
+        public System.Reflection.Assembly[] ReflectionOnlyGetAssemblies() { throw null; }
+        public void SetData(string name, object data) { }
+        //public void SetPrincipalPolicy(System.Security.Principal.PrincipalPolicy policy) { }
+        //public void SetThreadPrincipal(System.Security.Principal.IPrincipal principal) { }
+        public override string ToString() { throw null; }
+        public static void Unload(System.AppDomain domain) { }
+        public bool ShadowCopyFiles { get { throw null; } }
+        [Obsolete("AppDomain.AppendPrivatePath has been deprecated.")]
+        public void AppendPrivatePath(string path) { }
+        [Obsolete("AppDomain.ClearPrivatePath has been deprecated.")]
+        public void ClearPrivatePath() { }
+        [Obsolete("AppDomain.ClearShadowCopyPath has been deprecated.")]
+        public void ClearShadowCopyPath() { }
+        [Obsolete("AppDomain.SetCachePath has been deprecated.")]
+        public void SetCachePath(string path) { }
+        [Obsolete("AppDomain.SetShadowCopyFiles has been deprecated.")]
+        public void SetShadowCopyFiles() { }
+        [Obsolete("AppDomain.SetShadowCopyPath has been deprecated.")]
+        public void SetShadowCopyPath(string path) { }
+    }
+
+    public delegate System.Reflection.Assembly ResolveEventHandler(Object sender, ResolveEventArgs args);
+    public delegate void AssemblyLoadEventHandler(Object sender, AssemblyLoadEventArgs args);
+
+    public class AssemblyLoadEventArgs : EventArgs
+    {
+        public System.Reflection.Assembly LoadedAssembly { get { throw null;} }
+        public AssemblyLoadEventArgs(System.Reflection.Assembly loadedAssembly) { }
+    }
+
+    public partial class AppDomainUnloadedException : System.SystemException
```

Comment:
there are several places in vm where we throw AppDomainUnloadedException...but i guess since default domain is never unloaded it should never get thrown. So it should be OK to move the types to corefx.